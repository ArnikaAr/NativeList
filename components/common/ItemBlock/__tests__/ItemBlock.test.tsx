import React from 'react';
import renderer from 'react-test-renderer';
import ItemBlock from '../ItemBlock';
import { render } from '@testing-library/react-native';

const testItem = {
    id: '1',
    listName: 'List Name',
    listDetails: 'List Details',
    items: [{
        key: '1',
        value: 'test value'
    }]
};

describe('<ItemBlock />', () => {
    it('ItemBlock match snapshot', () => {
        const tree = renderer.create(<ItemBlock item={testItem} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('ItemBlock contain  right info', () => {
        const {getByText } = render(
            <ItemBlock item={testItem} />,
          );
          const listName = getByText("List Name");
          expect(listName).toBeTruthy();
    });
});

