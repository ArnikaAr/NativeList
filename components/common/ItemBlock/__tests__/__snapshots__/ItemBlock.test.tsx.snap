// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemBlock /> ItemBlock match snapshot 1`] = `
<View
  style={
    {
      "display": "flex",
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#d0d0d0",
        "borderBottomWidth": 2,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 15,
        "width": "100%",
      }
    }
  >
    <Text>
      Name
    </Text>
    <Text>
      List Name
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#d0d0d0",
        "borderBottomWidth": 2,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 15,
        "width": "100%",
      }
    }
  >
    <Text>
      List details
    </Text>
    <Text>
       
      List Details
    </Text>
  </View>
</View>
`;
